File: code/functions/rprism/R/load_ids.R
# @created: 30 Dec 21
# @modified: 30 Dec 21
# @author: Yoann Pradat
# 
# Exported functions
#     load_ids

#' Load identifiers files.
#'
#' This function loads the table linking all subject, sample and idspsy identifiers..
#'
#' @return a data.frame
#'
#' @param study Name of the cohort used in the naming of the files. Choose "met500", "prism" or "tcga".
#' @param ... Extra parameters passed to \code{\link{load_from_data}}
#'
#' @author Yoann Pradat
#'
#' @export
load_ids <- function(study, ...){
  df_ids <- load_from_data(get_filepath_ids_curated(study), progress=F, show_col_types=F, ...)

  df_ids
}

File: code/functions/pyprism/pyprism/data/_load_ids.py
# -*- coding: utf-8 -*-
"""
@created: 04/15/21
@modified: 04/15/21
@author: Yoann Pradat

    CentraleSupelec
    MICS laboratory
    9 rue Juliot Curie, Gif-Sur-Yvette, 91190 France

Load the curated tables of identifiers.
"""

import os
import pandas as pd

from ._util_load import load_from_data
from ._filepaths import _get_filepath_ids_curated

DataFrame = pd.core.frame.DataFrame

# User function ========================================================================================================

def load_ids(study: str):
    """
    Load clinical tables for the specified study.

    Parameters
    ----------
    study: str
        Choose one of 'met500, 'prism' or 'tcga'.

    Returns
    -------
    df_ids DataFrame
    """
    return load_from_data(_get_filepath_ids_curated(study), dtype="str")
